### PR TITLE
Enabling MFE and A-LTR for MSDK 2018R1 and above

### PR DESCRIPTION
**Pull request by taste1981**

### CODE REVIEW

File: talk/oms/sdk/base/win/msdkvideobase.cc [hunk 1]
```diff
@@ -2,34 +2,34 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include "msdkvideobase.h"
```
Comment (jianjunz) [inline talk/oms/sdk/base/win/msdkvideobase.cc]: The old order seems to be better.

File: talk/oms/sdk/base/win/msdkvideobase.cc [hunk 2]
```diff
@@ -40,22 +40,29 @@ bool MSDKFactory::Init() {
   return true;
 }
 
-MSDKFactory::~MSDKFactory() {
+MSDKFactory::~MSDKFactory() {}
+
+void MSDKFactory::SetMFETimeout(uint32_t timeout) {
```
Comment (jianjunz) [inline talk/oms/sdk/base/win/msdkvideobase.cc]: OMS C++ code usually use `type Property()` for getter and `void Property(type)` for setter.

File: talk/oms/sdk/base/win/msdkvideoencoder.cc [hunk 3]
```diff
@@ -190,20 +194,70 @@ int MSDKVideoEncoder::InitEncodeOnEncoderThread(const webrtc::VideoCodec* codec_
...
+    m_EncExtParams.push_back((mfxExtBuffer*)&multiFrameParam);
+
+    mfxExtMultiFrameControl multiFrameControl;
+    MSDK_ZERO_MEMORY(multiFrameControl);
+    multiFrameControl.Header.BufferId = MFX_EXTBUFF_MULTI_FRAME_CONTROL;
+    multiFrameControl.Header.BufferSz = sizeof(multiFrameControl);
+    multiFrameControl.Timeout = timeout;
+    m_EncExtParams.push_back((mfxExtBuffer*)&multiFrameControl);
+  }
+#endif
```
Comment (jianjunz) [inline talk/oms/sdk/base/win/msdkvideoencoder.cc]: I'm not quite familiar with this part. Maybe @HuaChunbo has more knowledge on it.

File: talk/oms/sdk/base/win/msdkvideoencoder.cc [hunk 4]
```diff
@@ -385,7 +455,7 @@ int MSDKVideoEncoder::Encode(
   }
 
   MSDK_ZERO_MEMORY(bs);
-  mfxU32 bsDataSize = param.mfx.FrameInfo.Width*param.mfx.FrameInfo.Height * 4;
```
Comment (jianjunz) [inline talk/oms/sdk/base/win/msdkvideoencoder.cc]: width*height*4 is not correct?

Comment (taste1981) [reply talk/oms/sdk/base/win/msdkvideoencoder.cc]: It's OK but we don't need to allocate that much bs buffer.

**Pull request merged**